src/lib.rs

```diff
@@ -356,7 +356,7 @@ pub fn narrowable_rboehm(args: TokenStream, input: TokenStream) -> TokenStream {
 
                 if t_vtable == vtable {
                     let objptr = unsafe { (self as *const _ as *const usize).add(1) };
-                    Some(unsafe { &*(objptr as *const U) })
+                    Some(unsafe { ::std::mem::transmute(objptr) })
```

Comment:
Why do we transmute here instead of using `Gc::from_raw`?

Reply:
Good question. I could use `Gc::from_raw` *except* its contract says "this pointer must have come from `Gc::into_raw`". I wonder if we want a `Gc::from_raw_interior` or something like that? At the moment it would behave identically but if we ever want to do something clever with interior pointers, it would give us that flexibility.

Reply:
Good point. `Gc::from_raw` would work fine here but technically is against the rules. I can't see a need for such a restriction on `from_raw` right now and have a feeling I may have imposed it a bit blindly while trying to be similar to `Rc`s API. I'll have a think about whether it's worth introducing a separate call to reconstruct a `Gc` from an interior pointer.

In the meantime, perhaps try using an `as` cast instead? It's a bit less dangerous than transmute and won't blindly compile if you ever decide to change the return type.

Reply:
Of the current options `from_raw` seems best, so I've done that in https://github.com/softdevteam/natrob/pull/14/commits/32841c70b54b37add6aa9ca2aa32c0e759ebe78f.